src/AdminConsole/Middleware/SecurityHeadersMiddleware.cs

```diff
@@ -0,0 +1,19 @@
+namespace Passwordless.AdminConsole.Middleware;
+
+public sealed class SecurityHeadersMiddleware
+{
+    private readonly RequestDelegate _next;
+
+    public SecurityHeadersMiddleware(RequestDelegate next)
+    {
+        _next = next;
+    }
+
+    public void Invoke(HttpContext context)
+    {
+        context.Response.Headers.Append("X-Content-Type-Options", new[] { "nosniff" });
```

Comment:
before-merging: Make sure you an admin can still download their API keys, this could potentially impact that.

Reply:
@abergs It was just moved from UseCsp.cs